tests/pstl/pstl_test_suite.hpp

```diff
@@ -58,4 +58,38 @@ struct non_trivial_copy {
   int x;
 };
 
+
+static thread_local int move_counter = 0;
+struct non_trivial_move {
+
+  non_trivial_move(){}
+
+  non_trivial_move(int val)
+  : x{val} {}
+
+  non_trivial_move(non_trivial_move&& other) {
+    x = std::move(other.x);
+    __acpp_if_target_host(++move_counter;)
+  }
+
+  non_trivial_move& operator=(non_trivial_move&& other) {
+    if (this != &other) {
+      x = std::move(other.x);
+      // x = std::exchange(other.x, {});
```

Comment:
What should the state of the moved object be? The standard mentions, "..the elements in the moved-from range will still contain valid values of the appropriate type, but not necessarily the same values as before the move". Using `std::exchange` would give the user more guarantees, but this implementation would then deviate from the standard by being more specific.

Reply:
I think for the purpose of this test, the code is fine (not sure if we even need a test case for non-trivial move, but it can't hurt)...The class doesn't do anything useful and is purely artifically made for the test, so I don't think it matters a lot how exactly it implements move assignment.